### ISSUE TITLE
UI hangs in Firefox when opening a project

### ISSUE THREAD

**Issue by jackyq2015** [OpenRefine/OpenRefine#956]
When open a project in firefox(windows 7), the left panel keep loading even the right panel is fully loaded. Under Chrome works fine!

Traced back to following commit by @DavidLeoni :
SHA-1: bf60360b26c76f39d41107654dfb4549813b79a6
- fixes language context in browser calls

Not sure it was caused by freebase or language context.

**Comment by tfmorris** [comment]
What version of OpenRefine is this?  Are there any error messages logged on the Javascript console?

**Comment by DavidLeoni** [comment]
I tried playing with Refine at that commit and Firefox 36.0.1 on my Windows 7, both with default and different context path, but couldn't reproduce the problem.

**Comment by jackyq2015** [comment]
Sorry forgot to mention that it works if it is deployed under windows 7 and access from 127.0.01. 

But if deploy under Linux and access from another windows 7  firefox, problem happens. It happens from the current master back to the the commit made by @DavidLeoni (maybe even more back but I have not tested more). But for the way of deployment, the following is good:
# good: [c68c1bb2b11a2c4894c0a7bbb7ec9b048a7367f4] Upgrade to Clojure 1.5.1 & switch to clojure-slim JAR - #792

I tried some bisect. But the result was confusing. I don't think it's correct. so I won't post here.

**Comment by jackyq2015** [comment]
Here is the output from the console running firefox, back end Linux. Looks the load-language are loaded again and again from some reason.

09:59:26.581 [                   refine] POST /command/core/load-language (4093ms)
09:59:26.970 [                   refine] POST /command/core/get-importing-configuration (389ms)
09:59:27.230 [                   refine] GET /command/core/get-all-project-metadata (260ms)
09:59:27.356 [                   refine] GET /command/core/get-languages (126ms)
09:59:27.453 [                   refine] GET /command/core/get-version (97ms)
09:59:30.610 [                   refine] POST /command/core/load-language (3157ms)
09:59:30.666 [                   refine] GET /command/core/get-preference (56ms)
09:59:30.773 [                   refine] GET /command/core/get-project-metadata (107ms)
09:59:30.803 [                  project] Loaded project 1783721362509 from disk in 0 sec(s) (30ms)
09:59:30.853 [                   refine] GET /command/core/get-models (50ms)
09:59:31.252 [                   refine] POST /command/core/get-rows (399ms)
09:59:47.015 [                   refine] POST /command/core/load-language (15763ms)			<--- why load again?
09:59:47.081 [                   refine] GET /command/core/get-preference (66ms)
09:59:48.163 [                   refine] GET /command/core/get-project-metadata (1082ms)
09:59:48.228 [                   refine] GET /command/core/get-models (65ms)
09:59:48.449 [                   refine] POST /command/core/get-rows (221ms)
09:59:48.903 [                   refine] POST /command/core/load-language (454ms)			<-- another time
09:59:48.968 [                   refine] GET /command/core/get-preference (65ms)
09:59:50.059 [                   refine] GET /command/core/get-project-metadata (1091ms)
09:59:50.126 [                   refine] GET /command/core/get-models (67ms)
09:59:50.343 [                   refine] POST /command/core/get-rows (217ms)
09:59:50.799 [                   refine] POST /command/core/load-language (456ms)		<-- another time
09:59:50.856 [                   refine] GET /command/core/get-preference (57ms)
09:59:51.954 [                   refine] GET /command/core/get-project-metadata (1098ms)
09:59:52.029 [                   refine] GET /command/core/get-models (75ms)
09:59:52.245 [                   refine] POST /command/core/get-rows (216ms)
09:59:52.675 [                   refine] POST /command/core/load-language (430ms)		<-- another time
09:59:52.834 [                   refine] GET /command/core/get-preference (159ms)
09:59:53.916 [                   refine] GET /command/core/get-project-metadata (1082ms)
09:59:54.034 [                   refine] GET /command/core/get-models (118ms)

@tfmorris, could you please take a look? From my bisect, it seems that this start from the Jquery upgrade. 4f2ebed676c201ca050e1dda0c1a9b561342ad5a is last good commit. I checked the previous one 094562e54b5c6a9af0df151f28996e860081bd47 which is a upgrade of JQuery.

**Comment by tfmorris** [comment]
Using Firefox 36.0.1 on Windows 7 with the current development version I, like @DavidLeoni,  can't reproduce this.  When it "hangs" is the browser spinning in a compute bound loop or is it idle?  What does Firebug show the browser doing?

There are definitely problems with 094562e54b5c6a9af0df151f28996e860081bd47 (typos in version numbers), but they are fixed by a later commit.

If there were problems with Firefox compatibility introduced in September, 2013, I'd be very surprised if it took 18 months for them to be discovered.

Another thing you could try is doing a hard browser refresh (or using a private browsing window) to make sure that you're not running stale cached code.

Let us know if you find any additional clues.

**Comment by jackyq2015** [comment]
It only happens when run it from Linux Server + Win 7 FF. If you run OpenRefine on Win 7 it works fine. Maybe that's the reason it hasn't be discovered for such long time. Also if you use chrome, it works fine.

when it "hangs" I can see the spinning at the left panel while the data got loaded at right panel. Firebug shows it tried to load the page and do "resize", calculating the padding etc(without exception). After running for a while, firefox will ask if you want to debug it since it won't response. But each time it stops at different place.  From the back end, I can see it keep getting "Get " ajax requests as I paste above.

I guess there might be some event is handled differently in FF + JQuery.

**Comment by tfmorris** [comment]
OK, thanks for the additional info.  I hadn't noticed the repeated requests in your log.  It's almost like it thinks it's restarting the open.

Of course, user installs always have the client and the server on the same machine, so they won't see this, but it'd still be good to track down since it probably hints at some underlying timing (or other) bug.  You may need to increase the logging level on the client and/or server to figure out *why* it thinks it needs to restart things.

**Comment by jackyq2015** [comment]
I tried to upgrade to latest version of jquery, jquery-ui, apache and firefox. The problem still there as long as I use the htpasswd authentication. Finally I roll back the jquery update to the old one( version before Sep 18, 2013), then It behave normally. 

For those who may come across the similar issue as me.

**Comment by tfmorris** [comment]
@jackyq2015 I'm trying to understand what the situation is here.  You closed the issue, but your last comment indicated that you had to fall back to an older version of jQuery.  Does the problem still exist in the current development sources?  Can you summarize what configuration(s) don't work?

**Comment by jackyq2015** [comment]
I closed it because it won't happen on current development sources. It only happens on our hosting environment which the deployment and access path is different(the OpenRefine core part is the same though). Our deployment environment is: Ubuntu + Apache + Basic Authentication. The problem only happens when the following combination is used:
1. Firefox (Windows or Linux)
2. passing the user name and password in the URL for Basic Authentication. If you type it, it won't happen

It took quite time to find out the problem. I will rather to treat it as a deployment related than the code based related. Since we will be upgrading the Apache Basic Authentication to SSL proxy, so I downgrade the version of JQuery for now.

**Comment by cbdavis** [comment]
I'm seeing what may be the same problem on openrefine-2.6-beta.1, openrefine-2.6-rc.2, and also a version built today from a fresh git checkout.  You can see a screenshot ![here](https://cloud.githubusercontent.com/assets/591006/11586413/19b35b60-9a73-11e5-9455-40c092c19865.png).

Under Chrome everything is indeed fine, and I've verified the problem on Ubuntu 15.10 running Firefox 42.0.  This may also cause confusion about projects not being saved.  In Firefox, clicking on 'Open Project' doesn't show anything even if you have created several projects, while viewing this in Chrome verifies that the projects are indeed there.

**Comment by jackyq2015** [comment]
Not sure if it's the same issue. My issue only happens on a OpenRefine behind a password-protected Apache server. You can go back to check your console to see if it is keeping loading some page or just hang somewhere.

But it seems Refine has more issue with FF than chrome.
